exporter/signalfxexporter/translation/delta_translator.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package translation
+
+import (
+	"github.com/gogo/protobuf/proto"
+	sfxpb "github.com/signalfx/com_signalfx_metrics_protobuf/model"
+)
+
+type deltaTranslator struct {
+	prevPts map[string]*sfxpb.DataPoint
+}
+
+func newDeltaTranslator() *deltaTranslator {
+	return &deltaTranslator{prevPts: map[string]*sfxpb.DataPoint{}}
+}
+
+func (t *deltaTranslator) translate(processedDataPoints []*sfxpb.DataPoint, tr Rule) []*sfxpb.DataPoint {
+	for _, currPt := range processedDataPoints {
+		deltaMetricName, ok := tr.Mapping[currPt.Metric]
+		if !ok {
+			// only metrics defined in Rule.Mapping get translated
+			continue
+		}
+		deltaPt := t.deltaPt(deltaMetricName, currPt)
+		if deltaPt == nil {
+			continue
+		}
+		processedDataPoints = append(processedDataPoints, deltaPt)
+	}
+	return processedDataPoints
+}
+
+func (t *deltaTranslator) deltaPt(deltaMetricName string, currPt *sfxpb.DataPoint) *sfxpb.DataPoint {
+	// check if we have a previous point for this metric + dimensions
+	dimKey := stringifyDimensions(currPt.Dimensions, nil)
+	fullKey := currPt.Metric + ":" + dimKey
+	prevPt, ok := t.prevPts[fullKey]
+	t.prevPts[fullKey] = currPt
```

Comment:
This looks good to me, but the structure will never be cleaned up from stale metrics in case of changed dimensions :( 
We need to look for a key-values storage with ttl

Reply:
here is the one promising thing I've run across https://stackoverflow.com/a/25487392

Reply:
Good idea -- I've replaced this with a ttl map.